src/EditorFeatures/Core/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs

```diff
@@ -63,17 +66,29 @@ private class BraceCompletionSession(
 
         public void Start()
         {
-            _threadingContext.ThrowIfNotOnUIThread();
+            ThreadingContext.ThrowIfNotOnUIThread();
 
-            // Brace completion is not cancellable.
-            var success = _threadingContext.JoinableTaskFactory.Run(() => TryStartAsync(CancellationToken.None));
-            if (!success)
+            // Brace completion is cancellable if the user has the 'responsive completion' option enabled.
+            var cancellationTokenSource = new CancellationTokenSource();
+            if (_timeoutThreshold > 0)
+                cancellationTokenSource.CancelAfter(_timeoutThreshold);
```

Comment:
consider something like to encapsulate the cancellationtokensource usage:

var cancellationToken = GetCancellationToken();
...

CancellationToken GetCancellationToken()
{
    if (_timeoutThreshold <= 0)
        return CancellationToken.None;

    var cancellationTokenSource = new CancellationTokenSource();
    cancellationTokenSource.CancelAfter(_timeoutThreshold);
    return cancellationTokenSource.Token;
}